src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -207,63 +207,127 @@ public Task<Either<ActionResult, DeleteReleasePlanViewModel>> GetDeleteReleaseVe
+                .OnSuccessDo(async release => await _cacheService.DeleteCacheFolderAsync(new PrivateReleaseContentFolderCacheKey(release.Id)))
                 .OnSuccessDo(async () => await _releaseDataFileService.DeleteAll(releaseVersionId))
                 .OnSuccessDo(async () => await _releaseFileService.DeleteAll(releaseVersionId))
                 .OnSuccessVoid(async releaseVersion =>
                 {
-                    // NOTE: As only removing unapproved amendments, releaseVersion.Release will not be orphaned
-                    // and associated item in releaseVersion.Publication.ReleaseSeries will also not be orphaned
-                    // so no need to remove them.
-
-                    releaseVersion.SoftDeleted = true;
-                    _context.ReleaseVersions.Update(releaseVersion);
-
-                    var roles = await _context
-                        .UserReleaseRoles
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    roles.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(roles);
-
-                    var invites = await _context
-                        .UserReleaseInvites
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    invites.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(invites);
-
-                    var methodologiesScheduledWithRelease =
-                        GetMethodologiesScheduledWithRelease(releaseVersionId);
-
-                    // TODO EES-2747 - this should be looked at to see how best to reuse similar "set to draft" logic
-                    // in MethodologyApprovalService.
-                    methodologiesScheduledWithRelease.ForEach(m =>
+                    if (!releaseVersion.Amendment && releaseVersion.ApprovalStatus == ReleaseApprovalStatus.Draft)
                     {
-                        m.PublishingStrategy = Immediately;
-                        m.Status = Draft;
-                        m.ScheduledWithReleaseVersion = null;
-                        m.ScheduledWithReleaseVersionId = null;
-                        m.Updated = DateTime.UtcNow;
-                    });
+                        await HardDeleteForDraft(releaseVersion, cancellationToken);
+                    }
+                    else
+                    {
+                        await SoftDeleteForAmendment(releaseVersion, cancellationToken);
+                    }
 
-                    _context.UpdateRange(methodologiesScheduledWithRelease);
+                    UpdateMethodologies(releaseVersionId);
 
                     await _context.SaveChangesAsync();
 
                     await _releaseSubjectRepository.DeleteAllReleaseSubjects(releaseVersionId: releaseVersionId);
```

Comment:
Again not your code, but this is mostly unnecessary - except for the clean up of orphaned `Subjects`. `ReleaseSubjects` and `ReleaseFootnotes` should be removed via cascade delete.

Happy for you to leave this - but would leave a comment if you don't.

Reply:
I'll leave a comment as discussed.